src/main.tsx

```diff
@@ -12,6 +12,74 @@ import { getSharingToken, isPublicShare } from '@nextcloud/sharing/public'
 import './viewer.css'
 import logger from './logger'
 
+declare global {
+	interface Window {
+		_nc_initial_state?: Map<string, unknown>
+	}
+}
+
+const normalizeCoreAppsInitialState = () => {
+	if (typeof window === 'undefined') {
+		return
+	}
+
+	const selector = '#initial-state-core-apps'
+	const mapKey = selector
+
+	const toArray = (value: unknown): Array<Record<string, unknown>> => {
```

Comment:
As mentioned in our call, I fear this can have some unexpected side effects for other cases where the initial state is used

This was introduced in https://github.com/nextcloud/server/pull/46370/files#diff-932e2df89d23d3a97ba9e651c29766107c29aef31566262cf1876107d7380637R77

Reply:
Actually https://github.com/nextcloud-libraries/nextcloud-vue/commit/d89450e94078f951b44e511b5ca7d3ef17278987 should have added a safeguard for this case

Reply:
Is not there in v9.0.1 which gets pulled in by @nextcloud/dialogs 

https://github.com/nextcloud-libraries/nextcloud-vue/blob/v9.0.1/src/components/NcAppContent/NcAppContent.vue#L170

Reply:
Maybe we need to downgrade @nextcloud/dialogs to the previous major